compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/DataflowAnalyzer.java

```diff
@@ -604,6 +613,17 @@ public void visit(BLangInvocation invocationExpr) {
         }
     }
 
+    private boolean isValidSelf(BSymbol symbol) {
```

Comment:
Shall we change the method signature to get the type (BObjectType) instead of symbol? 
Because symbol only used to getting the type here.

Reply:
Changed.